genesyscloud/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -886,47 +884,43 @@ func readUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{
 		}
 
 		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceArchitectUserPrompt())
-		if userPrompt.Name != nil {
-			d.Set("name", *userPrompt.Name)
-		} else {
-			d.Set("name", nil)
-		}
 
-		if userPrompt.Description != nil {
-			d.Set("description", *userPrompt.Description)
-		} else {
-			d.Set("description", nil)
-		}
+		resourcedata.SetNillableValue(d, "name", userPrompt.Name)
+		resourcedata.SetNillableValue(d, "description", userPrompt.Description)
 
-		if resources, ok := d.GetOk("resources"); ok && resources != nil {
-			promptResources := resources.(*schema.Set).List()
+		if resourcesSet, ok := d.Get("resources").(*schema.Set); ok && resourcesSet != nil {
+			promptResources := resourcesSet.List()
 			for _, promptResource := range promptResources {
-				resourceMap := promptResource.(map[string]interface{})
-				resourceFilename := resourceMap["filename"]
-				if resourceFilename.(string) == "" {
+				resourceMap, ok := promptResource.(map[string]interface{})
+				if !ok {
+					continue
+				}
+				resourceFilename, ok := resourceMap["filename"].(string)
+				if !ok || resourceFilename == "" {
 					continue
 				}
```

Comment:
This is part of the bug fix. This condition:
```golang
resourceFilename := resourceMap["filename"]
if resourceFilename.(string) == "" {
continue
}
```

was not working as expected because, if filename didn't exist in the map, <nil> would be returned, not an empty string. So the for loop would not "continue"